components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/util/APIErrorResponseHandler.java

```diff
@@ -0,0 +1,160 @@
+package org.wso2.carbon.identity.auth.valve.util;
+
+import org.apache.catalina.connector.Response;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.wso2.carbon.identity.auth.service.AuthenticationContext;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+import org.apache.log4j.MDC;
+
+public class APIErrorResponseHandler {
+
+    private static final Log log = LogFactory.getLog(APIErrorResponseHandler.class);
+
+    private static final String AUTH_HEADER_NAME = "WWW-Authenticate";
+    private static final String CORRELATION_ID_MDC = "Correlation-ID";
+
+    public static void handleErrorResponse(AuthenticationContext authenticationContext, Response response, int error,
```

Comment:
Doc comment is needed for all the public methods

Reply:
This has been fixed with 747c2c448717dcac5a09b0abec9838c5d61ac644